### DISCUSSION TITLE
Combine extensions with glob

### DISCUSSION THREAD

**Discussion by ryands17** [sharkdp/fd#1546] [Q&A] [answered]
Is it possible to combine the extension option with glob? So something like:

```sh
fd -e ts -g 'tests/**'
```

This doesn't work currently. Is there a better way?

**Comment by tavianator** [comment, marked as answer]
From the man page:

        -g, --glob
              Perform a glob-based search instead of a regular expression search.  If combined with the '--full-path' option,
              '**' can be used to match multiple path components.

So something like this should work:

```console
$ fd -e td -g '**/tests/**' --full-path
```

**Reply by ryands17** [reply]
Just an additional question: `**/tests/**` looks recursively in every folder. Is it possible to only check the parent folder and add a glob like this: `tests/**`. I tried this but it doesn't work.

**Reply by tavianator** [reply]
The glob itself does not affect the set of folders that `fd` looks in.  Think of the glob as just a pattern that is matched against the paths.  In this case we need the leading `**/` because the paths that `fd` matches with `--full-path` look like `/absolute/path/to/something/tests/foo.ts`.
